### ISSUE TITLE
Validate component IDs are unique

### ISSUE THREAD

**Issue by itowlson** [fermyon/spin#340]
Spin currently permits non-unique IDs, e.g. this loads without complaint:

```toml
[[component]]
id = "hello"
source = "target/wasm32-wasi/release/spinhelloworld.wasm"
[component.trigger]
route = "/hello"

[[component]]
id = "hello"
source = "target/wasm32-wasi/release/spinhelloworld.wasm"
[component.trigger]
route = "/hello"
```

There is currently stuff in the assets code that assumes component IDs are unique - we could fix this, but it might be better to just fail on duplicate ID, (1) because future code might make the same assumption and (2) it would avoid confusion during diagnostics.

**Comment by 0xcodeboi** [comment]
Hi @itowlson, I was taking a look at this and I _think_ it makes sense to add a uniqueness constraint somewhere around this code:
https://github.com/fermyon/spin/blob/5b0a48ecb9ea47563ae35f9ea414487fbc5365c5/crates/loader/src/bindle/mod.rs#L67-L74

My initial thought was that it probably makes more sense to have a constraint check in `config` crate, maybe using this error:
https://github.com/fermyon/spin/blob/5b0a48ecb9ea47563ae35f9ea414487fbc5365c5/crates/config/src/lib.rs#L16-L19

but the validation logic appears to be operating on a per-key basis and does not seem to have the full picture (i.e. it does not have a reference to _all_ the components) -- does this sound right or am I missing something?

I think there's a clear path forward to add a uniqueness check somewhere in the `loader` crate, if that sounds good I can take a stab at this 🙂

**Comment by radu-matei** [comment]
Fixed in #370
